app.js

```diff
@@ -41,6 +44,12 @@ app.use((err, req, res, next)=>{
             }
     next(err)
             })
+app.use((err, req, res, next)=>{
+    if(err.code === '23503'){
+    res.status(404).send({msg: "Id/foreign key not found"})
```

Comment:
'not found' is fine as an error message ... better to avoid giving any clues about the structure of your db